Task 3/my-app/routes/api.js

```diff
@@ -0,0 +1,67 @@
+const router = require('express').Router();
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const data = require('../helpers/data.json');
+
+router.post('/auth', (req, res) => {
+    const username = req.body.username;
+    const password = req.body.password;
+    if(Object.keys(data).includes(username)){
+        const auth_token = crypto.createHash('sha1').update(username+password).digest('hex');
+        res.cookie('username', username);
+        res.cookie('auth_token', auth_token);
+        res.status(302).header({Location: '/profile.html'}).send({});
+    } else {
+        res.status(302).header({Location: '/login.html'}).send({});
+    }
+})
+
+const auth = (req, res, next) => {
+    const cookies = req.cookies;
+    fs.readFile(path.join(__dirname, '../helpers/user.json'), (err, data) => {
+        if(err) throw err;
+        const password = (JSON.parse(data))[cookies.username].password;
+        const auth_token = crypto.createHash('sha1').update(cookies.username+password).digest('hex');
+        if(cookies.username && (cookies.auth_token === auth_token)) {
+            next();
+        } else {
+            res.status(302).send({"location": '/login.html'});
+        }
+    })
+};
```

Comment:
its a good practice to keep them under middleware directory instead of mixing them with routes.